infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSDeployer.java

```diff
@@ -0,0 +1,190 @@
+     * @return true or false depending on the result.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws TestGridInfrastructureException when an error occurs while reading the cf template file.
+     */
+    private boolean waitForAWSProcess(AmazonCloudFormation stackBuilder, String stackName) throws InterruptedException,
+            TestGridInfrastructureException {
+        DescribeStacksRequest wait = new DescribeStacksRequest();
+        wait.setStackName(stackName);
+        boolean completed = false;
+        String status = "UNKNOWN";
+        String reason = "";
+
+        log.info("Waiting ..");
+        while (!completed) {
+            List<Stack> stacks = stackBuilder.describeStacks(wait).getStacks();
+            if (stacks.isEmpty()) {
+                log.info("There is no stack by the name " + stackName);
+                return false;
+            } else {
+                for (Stack stack : stacks) {
+                    if (stack.getStackStatus().equals(StackStatus.CREATE_COMPLETE.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.DELETE_COMPLETE.toString())) {
+                        completed = true;
+                        return true;
+                    } else if (stack.getStackStatus().equals(StackStatus.CREATE_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_COMPLETE.toString())) {
+                        String errorMessage = "Error while executing infrastructure command due to :" +
+                                stack.getStackStatusReason();
+                        throw new TestGridInfrastructureException(errorMessage);
+                    }
+                }
+            }
+            if (!completed) {
+                Thread.sleep(5000);
+            }
+        }
+        log.info("Finished creating stack");
+        log.info(status + "(" + reason + ")");
+        return false;
+    }
+
+    /**
+     * Initialize the deployer with an infrastructure object.
+     *
+     * @param infrastructure infrastructure details.
+     */
+    public void init(Infrastructure infrastructure) {
+        this.infra = infrastructure;
+    }
+
+    /**
+     * This method destroys the CF infrastructure given the stack name.
+     *
+     * @param script Script object with the CloudFormaiton details.
+     * @return true if destroy finishes succesfully.
+     * @throws TestGridInfrastructureException when AWS error occurs in deletion process.
+     * @throws InterruptedException            when there is an interruption while waiting for the result.
+     */
+    public boolean destroyInfrastructure(Script script) throws TestGridInfrastructureException, InterruptedException {
+        String cfName = script.getName();
```

Comment:
Better rename the variable to `cloudFormationName` instead of `cfName` since it would indicate the purpose of this variable